raft-tech/TANF-app
# 2683-zap-CORS-misconfig

## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #2683 _

## How to Test
_List the steps to test the PR_

The test was done agains _raft_. CORS settings are applied to both staging and dev, so we should see same scan results against staging.

See below the scan result and note there is no CORS error:

![Screenshot 2023-10-18 at 11 53 19 AM](https://github.com/raft-tech/TANF-app/assets/97037188/064ea653-d448-40c3-aad5-84d2ad3814f5)


### Steps to test:

We can perform the scan from local machine by changing the scan address in ``` zap-scanner.sh```. Follow the steps below to perform scan from local:

1. Change [this line](https://github.com/raft-tech/TANF-app/blob/932a41ea991b83ac75ec5c93d187eb4785316d56/scripts/zap-scanner.sh#L22) to:
```
APP_URL="https://tdp-frontend-raft.app.cloud.gov"
```
2. Run the following command:
```
sh ./scripts/zap-scanner.sh frontend local dev
```


## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
